nuget/spec/dependabot/nuget/update_checker/compatibility_checker_spec.rb

```diff
@@ -3,19 +3,29 @@
 
 require "spec_helper"
 require "dependabot/dependency"
+require "dependabot/nuget/file_parser"
 require "dependabot/nuget/update_checker/compatibility_checker"
 require "dependabot/nuget/update_checker/repository_finder"
 require "dependabot/nuget/update_checker/tfm_finder"
 
 RSpec.describe Dependabot::Nuget::CompatibilityChecker do
   subject(:checker) do
+    Dependabot::Nuget::FileParser.new(dependency_files: dependency_files,
+                                      source: source,
+                                      repo_contents_path: repo_contents_path).parse
```

Comment:
Since several of the update_checker classes relied on information which is now supplied from the discovery.json, I run the FileParser prior to returning the class under test.